### PR TITLE
Fix ON CLUSTER queries hung for non leader replicas

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Bug Fix

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Fix bug when `ON CLUSTER` queries may hang forever for non-leader ReplicatedMergeTreeTables.

### CODE REVIEW

File: src/Interpreters/DDLWorker.cpp [hunk 1]
```diff
@@ -923,24 +931,45 @@ bool DDLWorker::tryExecuteQueryOnLeaderReplica(
...
 
     /// Not executed by leader so was not executed at all
     if (!executed_by_leader)
     {
-        task.execution_status = ExecutionStatus(ErrorCodes::NOT_IMPLEMENTED, "Cannot execute replicated DDL query");
+        /// If we failed with timeout
+        if (stopwatch.elapsedSeconds() >= MAX_EXECUTION_TIMEOUT_SEC)
+        {
+            LOG_WARNING(log, "Task {} was not executed by anyone, maximum timeout {} seconds exceeded", task.entry_name, MAX_EXECUTION_TIMEOUT_SEC);
+            task.execution_status = ExecutionStatus(ErrorCodes::TIMEOUT_EXCEEDED, "Cannot execute replicated DDL query, maximum retires exceeded");
```
Comment (tavplubix) [inline src/Interpreters/DDLWorker.cpp:966]:
```suggestion
            task.execution_status = ExecutionStatus(ErrorCodes::TIMEOUT_EXCEEDED, "Cannot execute replicated DDL query, timeout exceeded");
```

**Review by tavplubix — APPROVED** [review]

**Comment by alesapin** [comment]
No relevant errors.

**Comment by alesapin** [comment]
StorageRocksDB Failed one more time https://gist.github.com/alesapin/251a45e90e8fb87f26ec532b479badbf

**Comment by taiyang-li** [comment]
@alesapin I am curious about the reproduce process of the bug you had fixed, can you provide it?

**Pull request merged**